core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -434,27 +418,29 @@ private RefreshContext findRefreshIntervals(
             // For small sampling intervals such as '10T' or '2s' the actual sampler is
             // chosen as a 10x multiple of the original interval. That's to speed up
             // iteration done by the interval iterator.
-            final long minIntervalMicros = configuration.getMatViewMinRefreshInterval();
+            final long minInterval = driver.from(configuration.getMatViewMinRefreshInterval(), ColumnType.TIMESTAMP_MICRO);
             long approxBucketSize = timestampSampler.getApproxBucketSize();
             long actualInterval = viewDefinition.getSamplingInterval();
-            if (approxBucketSize < minIntervalMicros) {
-                while (approxBucketSize < minIntervalMicros) {
+            if (approxBucketSize < minInterval) {
+                while (approxBucketSize < minInterval) {
                     approxBucketSize *= 10;
                     actualInterval *= 10;
                 }
                 timestampSampler = TimestampSamplerFactory.getInstance(
+                        driver,
                         actualInterval,
                         viewDefinition.getSamplingIntervalUnit(),
                         0
                 );
             }
 
-            final long rowsPerBucket = estimateRowsPerBucket(baseTableReader, timestampSampler.getApproxBucketSize());
+            final long rowsPerBucket = estimateRowsPerBucket(driver, baseTableReader, timestampSampler.getApproxBucketSize());
```

Comment:
this is minor, but driver and sampler must be aligned on type. We either need an assert here or some sort of type safety. If they were to diverge - these problems will be hard to catch